src/app/components/editors/gameobject/gameobject-template/gameobject-template.integration.spec.ts

```diff
@@ -0,0 +1,161 @@
+
+    queryService = TestBed.get(QueryService);
+    querySpy = spyOn(queryService, 'query').and.returnValue(of());
+
+    spyOn(queryService, 'selectAll').and.returnValue(of(
+      { results: creatingNew ? [] : [originalEntity] }
+    ));
+
+    fixture = TestBed.createComponent(GameobjectTemplateComponent);
+    component = fixture.componentInstance;
+    page = new GameobjectTemplatePage(fixture);
+    fixture.autoDetectChanges(true);
+    fixture.detectChanges();
+  }
+
+  describe('Creating new', () => {
+    beforeEach(() => setup(true));
+
+    it('should correctly initialise', () => {
+      page.expectQuerySwitchToBeHidden();
+      page.expectFullQueryToBeShown();
+      page.expectFullQueryToContain(expectedFullCreateQuery);
+    });
+
+    it('changing a property and executing the query should correctly work', () => {
+      const expectedQuery = 'DELETE FROM `gameobject_template` WHERE (`entry` = ' + id + ');\n' +
+      'INSERT INTO `gameobject_template` (`entry`, `type`, `displayId`, `name`, `IconName`, `castBarCaption`, ' +
+      '`unk1`, `size`, `Data0`, `Data1`, `Data2`, `Data3`, `Data4`, `Data5`, `Data6`, `Data7`, `Data8`, `Data9`, ' +
+      '`Data10`, `Data11`, `Data12`, `Data13`, `Data14`, `Data15`, `Data16`, `Data17`, `Data18`, `Data19`, `Data20`, ' +
+      '`Data21`, `Data22`, `Data23`, `AIName`, `ScriptName`, `VerifiedBuild`) VALUES\n' +
+      '(' + id + ', 0, 0, \'Helias\', \'\', \'\', \'\', 1, 0, 0, 0, 0, 0, 0, 0, ' +
+      '0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, \'\', \'\', 0);';
+
+      querySpy.calls.reset();
+
+      page.setInputValueById('name', 'Helias');
+      page.clickExecuteQuery();
+
+      page.expectFullQueryToContain(expectedQuery);
+      expect(querySpy).toHaveBeenCalledTimes(1);
+      expect(querySpy.calls.mostRecent().args[0]).toContain(expectedQuery);
+    });
+  });
+
+  describe('Editing existing', () => {
+    beforeEach(() => setup(false));
+
+    it('should correctly initialise', () => {
+      page.expectDiffQueryToBeShown();
+      page.expectDiffQueryToBeEmpty();
+      page.expectFullQueryToContain(expectedFullCreateQuery);
+    });
+
+    it('changing all properties and executing the query should correctly work', () => {
+      const expectedQuery = 'UPDATE `gameobject_template` SET `displayId` = 1, `name` = \'2\', `IconName` = \'3\', ' +
+      '`castBarCaption` = \'4\', `unk1` = \'5\', `size` = 6, `Data0` = 7, `Data1` = 8, `Data2` = 9, `Data3` = 10, ' +
+      '`Data4` = 11, `Data5` = 12, `Data6` = 13, `Data7` = 14, `Data8` = 15, `Data9` = 16, `Data10` = 17, `Data11` = 18, ' +
+      '`Data12` = 19, `Data13` = 20, `Data14` = 21, `Data15` = 22, `Data16` = 23, `Data17` = 24, `Data18` = 25, ' +
+      '`Data19` = 26, `Data20` = 27, `Data21` = 28, `Data22` = 29, `Data23` = 30, `AIName` = \'31\', ' +
+      '`ScriptName` = \'32\' WHERE (`entry` = 1234);';
+
+      querySpy.calls.reset();
+
+      page.changeAllFields(originalEntity, ['entry', 'VerifiedBuild']);
```

Comment:
no need to manually exclude `entry`, since the way I've implemented `changeAllFields` is ignoring disabled fields by default ;)